guest/testdata/filter/main.go

```diff
@@ -133,3 +152,25 @@ func (postFilterPlugin) PostFilter(_ api.CycleState, pod proto.Pod, nodeMap api.
 		Reason: podSpecNodeName + " is unschedulable",
 	}
 }
+
+type preFilterExtensions struct{ noopPlugin }
+
+func (preFilterExtensions) AddPod(state api.CycleState, pod proto.Pod, podInfoToAdd api.PodInfo, nodeInfo api.NodeInfo) *api.Status {
+	nodeName := nodeInfo.Node().GetName()
+	state.Write(nodeName+pod.Spec().GetNodeName(), "ok")
+	status := 0
+	if nodeName == "bad" {
+		status = 1
+	}
+	return &api.Status{Code: api.StatusCode(status), Reason: "name is " + nodeName}
```

Comment:
You might need to confirm that podInfoToAdd is being passed to this function correctly like this:

```suggestion
	return &api.Status{Code: api.StatusCode(status), Reason: "Node name is " + nodeName + "and PodInfo name is " + podInfoToAdd.Pod().GetName()}
```